app/models/tag_layout.rb

```diff
@@ -12,9 +12,6 @@ class TagLayout < ApplicationRecord
 
   attr_accessor :tags_per_well
 
-  UnknownDirection = Struct.new(:direction)
-  UnknownWalking = Struct.new(:walking_by)
```

Comment:
@yoldas and @andrewsparkes these are the changes I made to TagLayout to prevent the exception that was happening in Sequencescape and causing an error 500 response in the API.  I tried a few things in the end but this felt the cleanest.  Effectively I am replacing the useless structs with dummy algorithms whose only job is to give a `nil` direction and a `nil` walking_by when they are called upon via the delegations on lines 71 and 72.  That way, the validation can be done against those attribute names and the error message makes more sense.  Otherwise you have to validate something the API user didn't provide a value for and the error response says something like "direction_algorithm - must not be blank".  With this pattern the error is "direction - must define a valid algorithm" and it handles all values that aren't from the hashes of algorithms above.